### PR TITLE
net: improve code documentation for dns seed behaviour

### PR DESCRIPTION
**Pull request by ajtowns**
Some better internal documentation post #16939

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
 /** Number of DNS seeds to query when the number of connections is low. */
 static constexpr int DNSSEEDS_TO_QUERY_AT_ONCE = 3;
 
+/** How long to delay before querying DNS seeds
+ *
+ * If we have more than THRESHOLD entries in addrman, then it's likely
+ * that we got those addresses from having previously connected to the p2p
```
Comment (laanwj) [inline src/net.cpp:57]: capitalize P2P please

File: src/net.cpp [hunk 2]
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
...
+ * network, and that we'll be able to successfully reconnect to the p2p
+ * network via contacting one of them. So if that's the case, spend a
+ * little longer trying to connect to known peers before querying the
+ * DNS seeds.
+ */
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11}; // 11sec
```
Comment (laanwj) [inline src/net.cpp:63]: the `11sec` comment is needless here, the constant (using `std::chrono` type is self-documenting)

File: src/net.cpp [hunk 3]
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
...
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_MANY_PEERS{300}; // 5min
```
Comment (laanwj) [inline src/net.cpp:64]: same but std::chrono::minutes(5)` ?

**Comment by fanquake** [comment]
Concept ACK. ecc86b9d514c6d67766eb2dd76f745ebfa36238a looks pretty good. Can you rebase and mark as ready for review?

**Comment by hebasto** [comment]
Concept ACK.

File: src/net.cpp [hunk 4]
```diff
@@ -52,10 +52,17 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
...
+ * network, and that we'll be able to successfully reconnect to the P2P
+ * network via contacting one of them. So if that's the case, spend a
+ * little longer trying to connect to known peers before querying the
+ * DNS seeds.
  */
-static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11}; // 11sec
-static constexpr std::chrono::seconds DNSSEEDS_DELAY_MANY_PEERS{300}; // 5min
-static constexpr int DNSSEEDS_DELAY_PEER_THRESHOLD = 1000; // "many" vs "few" peers -- you should only get this many if you've been on the live network
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11};
+static constexpr std::chrono::minutes DNSSEEDS_DELAY_MANY_PEERS{5};
```
Comment (MarcoFalke) [inline src/net.cpp:64]: ACK on this change

**Comment by naumenkogs** [comment]
ACK ecc86b9

**Comment by ajtowns** [comment]
Rebased, un-drafted PR, added some comments reflecting [review by @Sjors](https://github.com/bitcoin/bitcoin/pull/16939#pullrequestreview-420310676)

File: src/net.cpp [hunk 5]
```diff
@@ -53,10 +53,17 @@ static constexpr std::chrono::minutes DUMP_PEERS_INTERVAL{15};
...
+static constexpr int DNSSEEDS_DELAY_PEER_THRESHOLD = 1000; // "many" vs "few" peers
```
Comment (fanquake) [inline src/net.cpp:66]: nit: `"many" vs "few" peers` seems redundant now that there is an explainer above.

**Review by fanquake — APPROVED** [review]
ACK 5cb7ee67a5c0e5da39eb698b64d23722fb2f7b3e - thanks for following up.

File: src/net.cpp [hunk 6]
```diff
@@ -53,10 +53,17 @@ static constexpr std::chrono::minutes DUMP_PEERS_INTERVAL{15};
 static constexpr int DNSSEEDS_TO_QUERY_AT_ONCE = 3;
 
 /** How long to delay before querying DNS seeds
+ *
+ * If we have more than THRESHOLD entries in addrman, then it's likely
```
Comment (hebasto) [inline src/net.cpp:57]: nit: maybe
```suggestion
 * If we have more than PEER_THRESHOLD entries in addrman, then it's likely
```
?

**Review by hebasto — COMMENTED** [review]
ACK 5cb7ee67a5c0e5da39eb698b64d23722fb2f7b3e

File: src/net.cpp [hunk 7]
```diff
@@ -1797,6 +1809,9 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             return;
 
         // Add seed nodes if DNS seeds are all down (an infrastructure attack?).
+        // Note that we only do this if we started with an empty peers.dat,
+        // (in which case we will query DNS seeds immediately) *and* the DNS
+        // seeds have not returned any results.
```
Comment (ariard) [inline src/net.cpp:1814]: I don't think calling those peers "fixed seeds" matches with DNS seeds meaning. Contrary to DNS seeds open as `fOneShot=true`, we open regular connections to them, and not querying a `ADDR` and then disconnecting ?

**Review by ariard — COMMENTED** [review]
ACK 5cb7ee6

File: src/net.cpp (hunk 7 above)
Comment (naumenkogs) [reply src/net.cpp:1814]: I agree it's a bit confusing that we call both of them seeds, but the problem is not with this PR. This PR just adapts to the existing naming.
Are you suggesting renaming fixed seeds to hardcoded nodes or something?
That alone would require a separate PR.

**Comment by naumenkogs** [comment]
ACK 5cb7ee6

File: src/net.cpp (hunk 7 above)
Comment (ariard) [reply src/net.cpp:1814]: Yes I'm suggesting renaming them with a different names to mark difference of mechanism in a separate PR.

Comment (fanquake) [reply src/net.cpp:1814]: I'm going to merge this now. @ariard please open a followup for this (and if you want the other nits).

**Pull request merged**